lib/dry/struct/class_interface.rb

```diff
@@ -376,6 +376,12 @@ def |(type)
         end
       end
 
+      # @param meta [Boolean] Whether to dump the meta to the AST
+      # @return [Array] An AST representation
+      def to_ast(meta: true)
+        schema.to_ast(meta: meta)
+      end
```

Comment:
I believe we actually want a dedicated node type, maybe something like:

```ruby
[:struct, [name, [:schema, schema.to_ast(meta: meta)]]]
```

WDYT? /cc @flash-gordon

Reply:
of course, there's so much more to it that I simply didn't find a time to describe the details